src/containers/feedOfClaim/index.tsx

```diff
@@ -83,84 +85,99 @@ const SourceLink = ({ claim, searchTerm }: { claim: LocalClaim; searchTerm: stri
   )
 }
 
-const FeedClaim: React.FC<IHomeProps> = ({ toggleTheme, isDarkMode }) => {
-  const [claims, setClaims] = useState<Array<ImportedClaim>>([])
-  const [filteredClaims, setFilteredClaims] = useState<Array<ImportedClaim>>([])
-  const [visibleClaims, setVisibleClaims] = useState<Array<ImportedClaim>>([])
+async function fetchClaims(page: number, query?: string) {
+  const res = await axios.get(`${BACKEND_BASE_URL}/api/claimsfeed2`, {
+    timeout: 60000,
+    params: {
+      limit: PAGE_LIMIT,
+      search: query,
+      page
+    }
+  })
+  return res
+}
+
+const FeedClaim: React.FC<IHomeProps> = () => {
+  const [claims, setClaims] = useState<ImportedClaim[]>([])
+  const claimsRef = useRef<ImportedClaim[]>([])
+
   const location = useLocation()
-  const [isAuth, setIsAuth] = useState(false)
-  const [isLoading, setIsLoading] = useState(false)
+  // const [isAuth, setIsAuth] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
   const [selectedIndex, setSelectedIndex] = useState<null | number>(null)
-  const [searchTerm, setSearchTerm] = useState('')
-  const [showScrollButton, setShowScrollButton] = useState(false)
+  const [searchTerm, setSearchTerm] = useState(getSearchFromParams() || '')
   const navigate = useNavigate()
   const theme = useTheme()
   const isMediumScreen = useMediaQuery(theme.breakpoints.down('md'))
-  const isAuthenticated = checkAuth()
 
-  useEffect(() => {
+  const [showScrollButton, setShowScrollButton] = useState(false)
+  // const isAuthenticated = checkAuth()
```

Comment:
Is this needed? We can get rid of it if not needed and if it is, lets add a //**TODO** comment

Reply:
Removed it